packages/engine/Specs/Scene/I3SDataProviderSpec.js

```diff
@@ -168,6 +307,249 @@ describe("Scene/I3SDataProvider", function () {
     expect(testProvider.geoidTiledTerrainProvider).toEqual(geoidService);
   });
 
+  it("constructs I3SDataProvider with BSL", async function () {
+    spyOn(Resource.prototype, "fetchJson").and.callFake(function fetch() {
+      let mockedData;
+
+      if (this.url.includes("/sublayers/")) {
+        const id = this.url.match(/\/sublayers\/\d+\//)[0].split("/")[2];
+        mockedData = {
+          ...mockLayerData,
+          href: `layers/${id}/`,
+          id: parseInt(id),
+        };
+      } else {
+        mockedData = mockBSLProviderData;
+      }
+      return Promise.resolve(mockedData);
+    });
+    spyOn(Cesium3DTileset, "fromUrl").and.callFake(async () => {
```

Comment:
Throughout these units tests, I'm concerned this is mocking too much functionality for the test to be helpful.

Reply:
I3S functionality highly depends on external resources. To remove this dependency and make tests as simple as possible we mocked up the data and returns. This allows to split functionality to small parts and test each part separately. We tried to follow the style of existing I3S tests.